src/IO/readers.jl

```diff
@@ -0,0 +1,356 @@
+    cache[name] =
+        Topologies.Topology2D(Base.read!(reader, "meshes/" * mesh), elemorder)
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{SpectralElementSpace1D})
+
+Reads an object named 'name' of type 'SpectralElementSpace1D'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{SpectralElementSpace1D},
+)
+    (; file, cache) = reader
+    g = file["spaces/" * name]
+    quadrature_num_points = attrs(g)["quadrature_num_points"]
+    quadrature_style = eval(
+        Meta.parse(
+            attrs(g)["quadrature_type"] * "{$(quadrature_num_points)}()",
+        ),
+    )
+    topology = attrs(g)["topology"]
+    cache[name] = Spaces.SpectralElementSpace1D(
+        Base.read!(reader, "topologies/" * topology),
+        quadrature_style,
+    )
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{SpectralElementSpace2D})
+
+Reads an object named 'name' of type 'SpectralElementSpace2D'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{SpectralElementSpace2D},
+)
+    (; file, cache) = reader
+    g = file["spaces/" * name]
+    quadrature_num_points = attrs(g)["quadrature_num_points"]
+    quadrature_style = eval(
+        Meta.parse(
+            attrs(g)["quadrature_type"] * "{$(quadrature_num_points)}()",
+        ),
+    )
+    topology = attrs(g)["topology"]
+    cache[name] = Spaces.SpectralElementSpace2D(
+        Base.read!(reader, "topologies/" * topology),
+        quadrature_style,
+    )
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{ExtrudedFiniteDifferenceSpace})
+
+Reads an object named 'name' of type 'ExtrudedFiniteDifferenceSpace'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{T},
+) where {T <: ExtrudedFiniteDifferenceSpace}
+    (; file, cache) = reader
+    g = file["spaces/" * name]
+    staggering = eval(Meta.parse(attrs(g)["staggering"]))
+    vertical_topology = attrs(g)["vertical_topology"]
+    horizontal_space_name = attrs(g)["horizontal_space"]
+    vertical_space = Spaces.FiniteDifferenceSpace{staggering}(
+        Base.read!(reader, "topologies/" * vertical_topology),
+    )
+    horizontal_space = Base.read!(reader, "spaces/" * horizontal_space_name)
+    cache[name] =
+        Spaces.ExtrudedFiniteDifferenceSpace(horizontal_space, vertical_space)
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{Field})
+
+Reads an object named 'name' of type 'Field'.
+"""
+function read_new!(reader::HDF5Reader, name::AbstractString, ::Type{Field})
+    (; file, cache) = reader
+    @assert name ∈ keys(file["fields"])
+    g = file["fields/" * name]
+    data = read_dataset(g, "data")
+    data_layout = attrs(g)["data_layout"]
+    Nij = size(data, findfirst("I", data_layout)[1])
+    DataLayout = eval(Meta.parse(data_layout))
+    ElType =
+        eval(Meta.parse(replace(attrs(g)["value_type"], "ClimaCore." => "")))
```

Comment:
can we use an `if-else` branch instead of `eval`?

Reply:
`eval` statement has been replaced for `DataLayout`. `ElType` is much more complex to predict!